app/models/setting.rb

```diff
@@ -150,7 +149,6 @@ def parse_string_value(val)
         self.value = val.to_i
       else
         invalid_value_error _("must be integer")
-        return false
```

Comment:
These were returning `false` (a value which is not used), but never setting `self.value = `. As a consequence, `@setting` always had the same values as before, so `@setting.save` would always succeed.